integrations/datadog/client.py

```diff
@@ -0,0 +1,265 @@
+import json
+from typing import Any, AsyncGenerator, Optional
+from urllib.parse import urlparse, urlunparse
+import httpx
+
+from loguru import logger
+from port_ocean.context.event import event
+from port_ocean.utils import http_async_client
+
+MAX_PAGE_SIZE = 100
+
+
+class CacheKeys:
+    HOSTS = "_cache_hosts"
+    MONITORS = "_cache_monitors"
+    SLOS = "_cache_slos"
+    SERVICES = "_cache_services"
+
+
+def insert_credentials(url: str, username: str, password: str) -> str:
+    """
+    Inserts username and password into a given URL.
+
+    Args:
+        url (str): The original URL.
+        username (str): The username to insert.
+        password (str): The password to insert.
+
+    Returns:
+        str: The modified URL with inserted credentials.
+
+    Example:
+        new_url = insert_credentials("https://my.service.example.com", "my_username", "my_password")
+    """
+    parsed_url = urlparse(url)
+
+    # Insert credentials into the netloc part of the URL
+    netloc_with_credentials = f"{username}:{password}@{parsed_url.netloc}"
+
+    # Create a new URL with inserted credentials
+    modified_url = urlunparse(
+        (
+            parsed_url.scheme,
+            netloc_with_credentials,
+            parsed_url.path,
+            parsed_url.params,
+            parsed_url.query,
+            parsed_url.fragment,
+        )
+    )
+
+    return modified_url
```

Comment:
Datadog doesn't support passing in headers? can you share a link to the API documentation

Reply:
There are 2 options in the [docs](https://docs.datadoghq.com/integrations/webhooks/#authentication):

- Basic Auth
- Oauth 2

 This method creates the modified url that datadog parses to retrieve the basic auth credentials